### ISSUE TITLE
Incorrect violation for TrailingComma rule

### ISSUE THREAD

**Issue by nre-ableton** [CodeNarc/CodeNarc#374]
It seems that CodeNarc reports TrailingComma violations even when the list is not the thing that is spread on multiple lines. For example:

```groovy
void foo(List l) { /* stuff */ }
String bar(String a, String b, String c) { /* ... */ }

foo([bar(
  a,
  b,
  c,
)]) // <-- violation!
```

To work around the violation, the above code must be rewritten as such:

```groovy
foo([bar(
  a,
  b,
  c,
),])
```

I'm guessing that the reason this violation is thrown is that the list appears on multiple lines, but CodeNarc doesn't realize that it isn't the list which should have trailing commas in this case. I'm not sure if this is possible/easy to fix, but it would be nice since the above trailing comma looks a bit weird.

**Comment by chrismair** [comment]
Hmmm, I'm not really sure how to "fix" this. Technically the list does in fact span multiple lines, though in this case it is a single element that spans lines, not multiple list elements.  I am considering changing the rule to ignore lists/maps with single elements, if that is feasible.

**Comment by chrismair** [comment]
Thinking some more about this. It looks like it is possible to change the rule to ignore lists/maps with single elements. But I am wondering how useful that would be. Is it even worth it?

**Comment by darxriggs** [comment]
I think I hit the same issue.

My example: a list with a single element that is a map:
```
def list = [[
    a: 1,
    b: 2,
]]
```

Violation:
```
Violation[rule=TrailingCommaRule[name=TrailingComma, priority=3], lineNumber=1, sourceLine=def list = [[, message=List should contain trailing comma.]]
```

**Comment by nre-ableton** [comment]
@chrismair Ignoring lists/maps with a single element might be a good solution. I don't think it's ideal to solve this problem with an additional option, but I'm not sure I see a good way to fix the behavior seamlessly.

**Comment by chrismair** [comment]
The rule now ignores lists/maps with single elements by default. I added *ignoreSingleElementList* and *ignoreSingleElementMap* rule properties, defaulting to true.

Available with CodeNarc 1.3.

**Comment by nre-ableton** [comment]
@chrismair that seems like an excellent solution, thanks!
